docs/gameplay-matchmaker.md

```diff
@@ -1,182 +1,351 @@
 
-## Request opponents
+### Properties
 
-The server maintains a pool of users who've requested to be matched together with a specific number of opponents and each user's properties and filters if specified. Each user can add themselves to the matchmaker pool and register an event handler to be notified when enough users meet their criteria to be matched.
+Properties are key-value pairs that describe the user. Rank information, connecting region information, or selected match types are good examples of data that should be stored in properties. These properties are submitted by the user when they begin the matchmaking process, and may be different each time the user matchmakes.
 
-You can send a message to add the user to the matchmaker pool.
+When matchmaking completes these properties are visible to all matched users. You can store extra information without affecting the matchmaking process itself if it's useful to clients - just submit properties that aren't queried for as part of the matchmaking process.
 
 ```js fct_label="JavaScript"
 const message = { matchmaker_add: {
   min_count: 2,
-  max_count: 2,
-  query: "properties.a1:foo",
-  string_properties: {"a1": "bar"}
+  max_count: 4,
+  query: "*",
+  string_properties: {
+    region: "europe"
+  },
+  numeric_properties: {
+    rank: 8
+  }
 }};
-socket.send(message);
+var ticket = await socket.send(message);
 ```
 
-```csharp fct_label=".Net"
-// Updated example TBD
+```csharp fct_label=".NET"
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+var stringProperties = new Dictionary<string, string>(){
+  {"region", "europe"}
+};
+var numericProperties = new Dictionary<string, int>(){
+  {"rank", 8}
+};
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount, stringProperties, numericProperties);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
-```
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+var stringProperties = new Dictionary<string, string>(){
+  {"region", "europe"}
+};
+var numericProperties = new Dictionary<string, int>(){
+  {"rank", 8}
+};
 
-The message returns a ticket which can be used to cancel the matchmake attempt. A user can remove themselves from the pool if wanted. This is useful for some games where a user can cancel their action to matchmake at some later point and remove themselves being matched with other users.
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount, stringProperties, numericProperties);
+```
 
-Have a look at [this section](#rule-based-matchmaking) for more advanced matchmaking examples.
+### Query
 
-### Receive matchmake results
+The query defines how the user wants to find their opponents. Queries inspect the properties set by matchmaker users to find users eligible to be matched, or can ignore them to find any available users using the wildcard `"*"` query. A typical matchmaker query may look for opponents between given ranks, or in a particular region.
```

Comment:
Let's talk about the format of the `query` string. Is it space separated? How do you do multiple AND or OR queries?

Reply:
Are there any special symbols (beside `*`) that we need to mention here?